### PR TITLE
Add repo-commit field to metadata

### PR DESCRIPTION
**Pull request by krzyzacy**
So that testgrid will recognize and display the tag number.

### CODE REVIEW

File: jenkins/bootstrap.py [hunk 1]
```diff
@@ -478,6 +478,15 @@ def find_version(call):
         ]
         return call(cmd, output=True).strip()
 
+    # Make up some tag instead of unknown
+    cmd = ['git', 'describe', '--always', '--tags', '--abbrev=14']
```
Comment (ixdy) [inline jenkins/bootstrap.py]: if you want to always get a commit sha1, you probably want to use `git rev-parse --short=14 HEAD^{commit}` instead.

`git describe --always --tags` will use tags if they exist, e.g. on contrib right now:
```
$ git describe --always --tags --abbrev=14
cluster-autoscaler-0.4.0-470-g618f52dad58b52
```

Comment (krzyzacy) [inline jenkins/bootstrap.py]: ah thank! Was only trying in test-infra

File: jenkins/bootstrap.py [hunk 2]
```diff
@@ -478,6 +478,15 @@ def find_version(call):
...
+    try:
+        tag = call(cmd, output=True)
+        if tag:
+            return '+%s' % tag.strip()
```
Comment (ixdy) [inline jenkins/bootstrap.py]: also, I think the output of this function is intended to be a semantic version, which this is not.

@rmmh will this work?

Comment (krzyzacy) [inline jenkins/bootstrap.py]: gubernator always look for k/k and will not anyway...

we can somehow append the repo name here and try to fix the link?

**Comment by rmmh** [comment]
Let's put the repo's commit in finished.json instead, and make testgrid use that if available.

File: jenkins/bootstrap.py (hunk 2 above)
Comment (ixdy) [inline jenkins/bootstrap.py]: oh right, you mentioned testgrid. does it do the right thing here?

alternately: we already save `repo` in the metadata section of `finished.json` - maybe we should have a new field `git-commit` too?

which is what @rmmh just said

**Comment by krzyzacy** [comment]
Oh I was hoping I can test it from the PR but guess PR uses bootstrap from master :crab:

**Comment by krzyzacy** [comment]
Will make a testgrid change to look for this commit number

File: jenkins/bootstrap.py [hunk 3]
```diff
@@ -370,7 +370,7 @@ def append_result(gsutil, path, build, version, passed):
         errors += 1
 
 
-def metadata(repo, artifacts):
+def metadata(repo, artifacts, call):
```
Comment (rmmh) [inline jenkins/bootstrap.py]: Don't do this. Use `Stub(bootstrap, 'call', FakeCall())`.

File: jenkins/bootstrap.py [hunk 4]
```diff
@@ -385,10 +385,17 @@ def metadata(repo, artifacts):
     if not meta or not isinstance(meta, dict):
         meta = {}
     meta['repo'] = repo
+
+    try:
+        commit = call(['git', 'rev-parse', '--short=14', 'HEAD^{commit}'], output=True)
```
Comment (rmmh) [inline jenkins/bootstrap.py]: How is "HEAD^{commit}" different from HEAD? We might as well use the full commit hash here (no --short)-- you can truncate it for testgrid display.

Comment (krzyzacy) [inline jenkins/bootstrap.py]: hummm... seems give the same value and Google does not wanna tell me the difference... @ixdy

Comment (krzyzacy) [inline jenkins/bootstrap.py]: I saw two comments but github only displayed one? @rmmh did you delete the other one?

Comment (ixdy) [inline jenkins/bootstrap.py]: I cargo-culted this from elsewhere, but it probably doesn't matter for something like `HEAD`. It is more explicit about resolving to return a commit sha1, however.

From the [docs](https://www.kernel.org/pub/software/scm/git/docs/gitrevisions.html#_specifying_revisions):
> <rev>^{<type>}, e.g. v0.99.8^{commit}
A suffix ^ followed by an object type name enclosed in brace pair means dereference the object at <rev> recursively until an object of type <type> is found or the object cannot be dereferenced anymore (in which case, barf). For example, if <rev> is a commit-ish, <rev>^{commit} describes the corresponding commit object. Similarly, if <rev> is a tree-ish, <rev>^{tree} describes the corresponding tree object. <rev>^0 is a short-hand for <rev>^{commit}.

It looks like it matters more if we were dereferencing a tag. e.g. on k/k:
```console
$ git rev-parse v1.5.3
2d49a2edd9e387873da2e5b5dbe74d61c9b95665
$ git rev-parse v1.5.3^{commit}
029c3a408176b55c30846f0faedf56aae5992e9
$ git rev-parse v1.5.3^{object}
2d49a2edd9e387873da2e5b5dbe74d61c9b95665
```

I think 2d49... is referring to the tag object, whereas 029c... is referring to the actual commit pointed to by the tag `v1.5.3`.

File: jenkins/bootstrap.py [hunk 5]
```diff
@@ -385,10 +385,17 @@ def metadata(repo, artifacts):
     if not meta or not isinstance(meta, dict):
         meta = {}
     meta['repo'] = repo
+
+    try:
+        commit = call(['git', 'rev-parse', 'HEAD^{commit}'], output=True)
```
Comment (ixdy) [inline jenkins/bootstrap.py]: I think you want to `.strip()` this too

Comment (krzyzacy) [inline jenkins/bootstrap.py]: stripped in the underneath line.

**Comment by krzyzacy** [comment]
PTAL

**Comment by rmmh** [comment]
Can you add a check under IntegrationTest that git-commit is filled in with a revision of the expected length? That will catch bad git commands.

**Comment by krzyzacy** [comment]
@rmmh something like this?

**Comment by krzyzacy** [comment]
screwed, 1sec

**Comment by krzyzacy** [comment]
descrewed.. was moving it outside IntegrationTest..

@rmmh PTAL

**Pull request merged**